proxy/coco/src/peer.rs

```diff
@@ -323,6 +339,24 @@ impl Api {
         callback(&mut browser)
     }
 
+    /// Get the underlying [`librad::net::protocol::Protocol`].
+    ///
+    /// # Errors
+    ///
+    /// * if they `callback` errors
+    pub fn with_protocol<F, T>(&self, callback: F) -> BoxFuture<'static, Result<T, Error>>
```

Comment:
Why `BoxFuture`, actually? A generic parameter should work fine, and then the caller can choose to box or not.

Reply:
If we don't box here, the lock guard we hold for the lock we have around the internal API is crossing await boundaries. Didn't find a better way to solve that, open to suggestions.